File: linux_ui/src/lib/TerminalResponse.ts
export interface TerminalEntry {

    path: string;

    command: string;

    output: string;

}


export interface TerminalResponse {

    mode:
        | "shell"
        | "editor"
        | "python";


    output: string;


    currentPath?: string;


    history?: TerminalEntry[];


    file?: string;


    language?: string;


    content?: string;


}

File: WebLinux.Kernel/Core/TerminalResponse.cs
namespace WebLinux.Kernel.Core;

public class TerminalResponse
{
    public string Mode { get; set; } = "shell";
    public string Output { get; set; } = "";
    public string CurrentPath { get; set; } = "~";

    public List<TerminalEntry> History { get; set; } = new();
    
    public string? File { get; set; }
    public string? Language { get; set; }
    public string? Content { get; set; }
}
